src/sentry/issues/auto_source_code_config/code_mapping.py

```diff
@@ -593,9 +593,12 @@ def get_path_from_module(module: str, abs_path: str) -> tuple[str, str]:
     if "." not in module:
         raise DoesNotFollowJavaPackageNamingConvention
 
-    # If module has a dot, take everything before the last dot
```

Comment:
These changes simplify this code block.

It also fixes a bug where we were not restricting the length of the stack root, thus, we would end up with `com/example/` and many more levels deep (not intended behaviour).

There's a test down below that ensures we get the right number of levels.

Inputs:
* repo_file="src/com/example/foo/bar/Baz.kt"
* self.frame(module="com.example.foo.bar.Baz", abs_path="Baz.kt", in_app=False)

Expected:
* stack_root="com/example/", source_root="src/com/example/"
* in_app_rule="stack.module:com.example.** +app"